src/module/actor/spellcasting.ts

```diff
@@ -24,8 +24,7 @@ export class ActorSpellcasting extends Collection<SpellcastingEntryPF2e> {
         return this.regular.filter((entry) => entry.isPrepared || entry.isSpontaneous);
     }
 
-    canCastConsumable(item: ConsumablePF2e): boolean {
-        const spell = item.embeddedSpell;
+    canCastConsumable(spell?: SpellPF2e): boolean {
```

Comment:
canCastConsumable(spell) feels off to me lmao. That said, this method was here to also in the future be able to handle things like scroll thaumaturgy. I'd continue to give it the consumable, make the method async, but cache the embedded spell (cache wiped during data prep).